### ISSUE TITLE
Implement infinite scrolling list for results on mobile version 

### ISSUE THREAD

**Issue by KaiDrieHda** [hochschule-darmstadt/einander-helfen#191]
**Reason (Why?)**
The usability of scrolling through the results on the mobile version shall be improved.

**Solution (What?)**
The list showing the results shall be infinite scrolling on the mobile version. When going to the result page on the mobile version, 20 items shall be loaded. When scrolling to the bottom of that list, there shall be a button "Show more/Mehr anzeigen", that loads 20 more items of the list and adds them to the bottom of the list so far.
As soon as the user scrolls the list, an arrow button will appear, allowing the user to jump back to the top of the list. 

**Out of scope**
Changes to the map, the results page in web view and any other page that is not the result page.

**Relation to other Issues**
#122 

**Acceptance criteria**

1. The list is infinite scrolling.
2. Only 20 items are shown.
3. There is a "Show more/Mehr anzeigen"-button that loads 20 more items.
4. There is an arrow button that jumps to the top of the list.

**Comment by BernhardHumm** [comment]
Should there be a button “mehr anzeigen” or should the next page be loaded automatically when reaching the end of the list?

@ fachliche Architekten: What is expected behaviour in smartphone apps?

 

Is result size of 10 a good choice? Wouldn’t 20 be more suitable? @ fachliche Architekten: What is common in smartphone apps?

 

 

 

Von: LeDrieHda <notifications@github.com> 
Gesendet: Montag, 30. November 2020 17:48
An: hochschule-darmstadt/einander-helfen <einander-helfen@noreply.github.com>
Cc: Subscribed <subscribed@noreply.github.com>
Betreff: [hochschule-darmstadt/einander-helfen] Implement infinite scrolling list for results on mobile version (#191)

 

Reason (Why?)
The usability of scrolling through the results on the mobile version shall be improved.

Solution (What?)
The list showing the results shall be infinite scrolling on the mobile version. When going to the result page on the mobile version, 10 items shall be loaded. When scrolling to the bottom of that list, there shall be a button "Show more/Mehr anzeigen", that loads 10 more items of the list and adds them to the bottom of the list so far.
As soon as the user scrolls the list, an arrow button will appear, allowing the user to jump back to the top of the list.

Out of scope
Changes to the map, the results page in web view and any other page that is not the result page.

Relation to other Issues
#122 <https://github.com/hochschule-darmstadt/einander-helfen/issues/122> 

Acceptance criteria

1.	The list is infinite scrolling.
2.	Only 10 items are shown.
3.	There is a "Show more/Mehr anzeigen"-button that loads 10 more items.
4.	There is an arrow button that jumps to the top of the list.

**Comment by LeDrieHda** [comment]
We decided against loading the page automatically once you reach the end of the page, because otherwise you will never be able to reach the footer. Because we have little space on the mobile view and the header is already quite big, we decided against making the footer sticky and for the "Mehr anzeigen" button.

We discussed the result size of 10 and tried out the look and feel of both 10 and 20 results on the mobile view, which were both acceptable. We checked other websites/apps for the size they implemented; apart from Google who also has 10, most other sites show more. Also it would be preferable to not have to click on the "Mehr anzeigen"-button every 10 results, which is why we decided for changing it to 20.

**Comment by BernhardHumm** [comment]
OK, thanks

 

 

 

 

Von: LeDrieHda <notifications@github.com> 
Gesendet: Mittwoch, 2. Dezember 2020 20:01
An: hochschule-darmstadt/einander-helfen <einander-helfen@noreply.github.com>
Cc: Humm, Bernhard, Prof. Dr. <bernhard.humm@h-da.de>; Comment <comment@noreply.github.com>
Betreff: Re: [hochschule-darmstadt/einander-helfen] Implement infinite scrolling list for results on mobile version (#191)

 

We decided against loading the page automatically once you reach the end of the page, because otherwise you will never be able to reach the footer. Because we have little space on the mobile view and the header is already quite big, we decided against making the footer sticky and for the "Mehr anzeigen" button.

We discussed the result size of 10 and tried out the look and feel of both 10 and 20 results on the mobile view, which were both acceptable. We checked other websites/apps for the size they implemented; apart from Google who also has 10, most other sites show more. Also it would be preferable to not have to click on the "Mehr anzeigen"-button every 10 results, which is why we decided for changing it to 20.
